### ISSUE TITLE
Support for ReactNative

### ISSUE THREAD

**Issue by giautm** [jdiamond/MQTT.ts#9]
Hi, thank you for build this library.

Currently, don't have any library good enough to use MQTT in ReactNative. Some libraries are using native libraries (Object-C/Java) to implement then wrap APIs in Javascript (MQTT over TCP: https://github.com/davesters/rn-native-mqtt). Some libraries are fork from the unmaintained library and didn't update anymore (MQTT over Websocket: https://github.com/rh389/react-native-paho-mqtt).

Until I found your work!

Support RN is simple, they have WebSocket API implement. Only a few things need to do, implement UTF Encoder/Decoder, and URL polyfill (react-native-url-polyfill)

**Comment by giautm** [comment]
Hi @jdiamond , any comment?

**Comment by jdiamond** [comment]
I'm open to it, but I don't have experience with React Native to really know what you need. Not sure if I would add a new build to the existing npm package or publish a new one?

**Comment by giautm** [comment]
Yes, I really have a working version here: https://github.com/giautm/mqtt-native/tree/main/src (Just for test).

ReactNative can compile typescript directly without bundle first. 

I need to add `import { URL } from 'react-native-url-polyfill'` as first line in https://github.com/giautm/mqtt-native/blob/main/src/client/base_client.ts

And polyfill for TextEncoder/TextDecoder.

**Comment by jdiamond** [comment]
Thanks, I will study it this weekend.

**Comment by giautm** [comment]
Yeah, Your library working very well with ReactNative. I have a client that connects to a broker over 5 days without any interruption/disconnect. My application working stable for 5 days. 👍 

But, I don't have any knowledge of the MQTT protocol. So I can't maintain it well, I can help to maintain a RN support package. :D
